tests/utils.py

```diff
@@ -241,3 +244,55 @@ def teardown_spark():
 
 def _scala_test_jar_url():
     return s3.http_url(os.path.basename(os.environ["SCALA_TEST_JAR_PATH"]))
+
+
+def spark_security_session():
+    '''
+    Spark strict mode setup is slightly different from dcos-commons, so can't use sdk_security::security_session.
+    '''
+    user = 'nobody'
+    role = '*'
+    service_account = SPARK_SERVICE_ACCOUNT
+    secret = SPARK_SERVICE_ACCOUNT_SECRET
+
+    def grant_driver_permission(service_account_name, app_id):
+        dcosurl, headers = sdk_security.get_dcos_credentials()
+        app_id_encoded = urllib.parse.quote(app_id, safe='')
+        app_id_encoded = urllib.parse.quote(app_id_encoded, safe='')  # double-encoded (why?)
```

Comment:
could you nest these so there isn't two assignments?

Reply:
Sure